### PR TITLE
[gmp] Fix building for linux x86

### PR DESCRIPTION
**Pull request by robbert-ef**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by robbert-ef** [comment]
@microsoft-github-policy-service agree company="Easyflex"

File: ports/gmp/portfile.cmake [hunk 1]
```diff
@@ -65,6 +65,10 @@ if(VCPKG_DETECTED_CMAKE_C_COMPILER_ID STREQUAL "MSVC")
 elseif(VCPKG_TARGET_IS_MINGW AND VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
     # not exporting asm functions
     set(disable_assembly ON)
+elseif(VCPKG_TARGET_IS_LINUX AND VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
+    set(ccas "${VCPKG_DETECTED_CMAKE_C_COMPILER}")
+    vcpkg_list(APPEND OPTIONS "ABI=32")
+    string(APPEND asmflags " -m32")
```
Comment (dg0yt) [inline ports/gmp/portfile.cmake:71]: I have recently seen/used the same x86 pattern in another port but for Android. 
- Do we really need `ABI=32` in addition to `-m32`?
- Which other x86 toolchains/triplets will understand `-m32`?
- Can we get `-m32` from the cmake toolchain?

Comment (robbert-ef) [reply ports/gmp/portfile.cmake:71]: `ABI=32` is required, fixes:
```
configure: error: Oops, mp_limb_t is 32 bits, but the assembler code
in this configuration expects 64 bits.
You appear to have set $CFLAGS, perhaps you also need to tell GMP the
intended ABI, see "ABI and ISA" in the manual.
```
The `-m32` is also necessary, without it, compile of asm fails:
```
tmp-add_n.s: Assembler messages:
tmp-add_n.s:202: Error: operand type mismatch for `jmp'
make[2]: *** [add_n.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
tmp-sub_n.s: Assembler messages:
tmp-sub_n.s:202: Error: operand type mismatch for `jmp'
make[2]: *** [sub_n.lo] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

Comment (Neumann-A) [reply ports/gmp/portfile.cmake:71]: If you need asm flags you might want to actually request it from `vcpkg_get_cmake_vars`.

Comment (dg0yt) [reply ports/gmp/portfile.cmake:71]: Unfortunately, `vcpkg_cmake_get_vars` isn't taking parameters such as `LANGUAGES` ATM. (And maybe it should be handled gracefully if ASM can't be enabled.)

**Pull request merged**